src/ValueResolver/NodeExpressionResolver.php

```diff
@@ -232,7 +232,7 @@ protected function resolveExprConstFetch(Expr\ConstFetch $node)
         if (!$isFQNConstant) {
             if (method_exists($this->context, 'getFileName')) {
                 $fileName      = $this->context->getFileName();
-                $namespaceName = $this->context->getNamespaceName();
+                $namespaceName = $this->resolveScalarMagicConstNamespace();
```

Comment:
Method `resolveScalarMagicConstNamespace()` is used dynamically for nodes of `\PhpParser\Node\Scalar\MagicConst\Namespace_` class (that holds \_\_NAMESPACE__ magic const).

However, I can agree that we can reuse it for our needs. So, :+1: